lib/core/src/Cardano/Pool/DB/Sqlite.hs

```diff
@@ -208,14 +211,20 @@ newDBLayer trace fp timeInterpreter = do
 
             pure (foldl' toMap Map.empty production)
 
-        readTotalProduction = do
-            production <- fmap entityVal <$>
-                selectList ([] :: [Filter PoolProduction]) []
-
-            let toMap m (PoolProduction{poolProductionPoolId}) =
-                    Map.insertWith (+) poolProductionPoolId 1 m
-
-            pure $ Map.map Quantity $ foldl' toMap Map.empty production
+        readTotalProduction =
+            Map.fromList . rights . fmap parseRow <$> rawSql query []
```

Comment:
This looks like we hide parse errors?

Reply:
> This looks like we hide parse errors?

It does indeed!

I previously added a patch that would cause this function to fail fast in the face of a parse error. However, that had the disadvantage that a single row failing to parse would cause the entire operation to fail.

After giving this some thought, I think the following method would be more appropriate, especially from a user experience perspective:

- for rows that fail to parse: create a log entry for each error.
- for rows that parse successfully: return these as part of the result.

Reply:
Fixed in 6fbc807c47cfea70f3f7c7e4fa628048cc703ac6.